spec/controllers/manage_assessments/attachments_controller_spec.rb

```diff
@@ -0,0 +1,20 @@
+require "rails_helper"
+
+describe ManageAssessments::AttachmentsController do
+  describe "GET #show" do
+    it "downloads the requested attachment from server" do
+      course = create(:course)
+      outcome = create(:outcome)
+      coverage = create(:coverage, course: course, outcomes: [outcome])
+      outcome_coverage = OutcomeCoverage.first
+      assignment = create(:assignment, outcome_coverage: outcome_coverage)
```

Comment:
I have simplified some of this setup with factories in my results branch which may make sense to include here to clean this up. Try these out:

```ruby
  factory :assignment do
    name
    problem "problem 1"

    after(:build) do |assignment|
      if assignment.outcome_coverage.nil?
        coverage = build(:coverage)
        assignment.outcome_coverage = coverage.outcome_coverages.first
      end
    end
  end
```

```ruby
  factory :coverage do
    transient do
      outcomes []
    end

    course
    subject

    after(:build) do |coverage, evaluator|
      if evaluator.outcomes.present?
        coverage.outcomes = Array(evaluator.outcomes)
      else
        coverage.outcomes = [
          build(:outcome, course: coverage.course)
        ]
      end
    end
  end
```

If you do this, I think you'll be able to say `create(:assignment)` and get an assignment where the various relationships all talk about the same things.

Do you want to try these out and see if they work for you?